ussologin/visitwebpage.go

```diff
@@ -26,28 +26,24 @@ func VisitWebPage(filler form.Filler, client *http.Client, store TokenStore) fun
 	return func(u *url.URL) error {
 		lm, err := idmclient.LoginMethods(client, u)
 		if err != nil {
-			return err
+			return errgo.Mask(err)
 		}
 		if lm.UbuntuSSOOAuth != "" {
 			var tok *usso.SSOData
 			var err error
-			if store == nil {
-				tok, err := Login(filler)
-				if err != nil {
-					return err
-				}
-				return doSignedRequest(client, lm.UbuntuSSOOAuth, tok, u)
+			if store != nil {
+				tok, _ = store.Get()
 			}
-			if tok, err = store.Get(); err != nil {
-				tok, err = Login(filler)
+			if tok == nil {
+				tok, err = GetToken(filler, "charm")
 				if err != nil {
-					return err
+					return errgo.Mask(err)
 				}
-				if err := store.Put(tok); err != nil {
-					return errgo.Notef(err, "cannot save token")
+				if store != nil {
+					store.Put(tok)
```

Comment:
Is it really OK to ignore the error here? If so, a comment justification would be appropriate, I think.